src/server/hset_family.cc

```diff
@@ -579,6 +579,16 @@ OpResult<vector<string>> OpGetAll(const OpArgs& op_args, string_view key, uint8_
     }
   }
 
+  // Empty hashmaps must be deleted, this case only triggers for expired values
+  // and the enconding is guranteed to be a DenseSet since we only support expiring
+  // value with that enconding.
+  if (res.empty()) {
+    auto mutable_res = db_slice.FindMutable(op_args.db_cntx, key, OBJ_HASH);
```

Comment:
I dispatch another `Find` operation because:

1.  I do not want to break `constness` above
2. This is a rather exceptional case so the second `Find` is fine

Reply:
Also can this `Find` fail because of the whole `Hashmap` being expired (not the individual fields which are checked while iterating over the dense set)

Reply:
it can not happen, we use transaction time here